src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectHandler.cs

```diff
@@ -648,6 +679,12 @@ public OpenIdConnectHandler(HttpClient backchannel)
             ticket = userInformationReceivedContext.AuthenticationTicket;
             user = userInformationReceivedContext.User;
 
+            Options.ProtocolValidator.ValidateUserInfoResponse(new OpenIdConnectProtocolValidationContext()
+            {
+                ClientId = Options.ClientId,
+                ProtocolMessage = message,
```

Comment:
This is not correct. You need to set 2 properties:
1. UserInfoEndpointResponse: which is of type JwtSecurityToken
2. ValidatedIdToken: which is the validated token that is used to create the identity (JwtSecurityToken corresponding to message.idToken

All that validator does is match the "sub" claim. Also, you can get rid of the "sub" claim matching logic below.